<% menu_section='VISUALS' %>
<% include ../add_ins/begin.ejs %>
<% var info = JSON.parse(proj_info) %>
<% var env_sources = JSON.parse(md_env_package) %>
<% var targets = JSON.parse(constants).TARGETS %>
<% var portals = JSON.parse(constants).PORTALS %>
<% if(filtering == undefined){ %>
<%  filtering = 0 %>
<% } %>
<script defer>
  var datasets_local  = JSON.parse('<%- data_to_open %>');
  var portal_local  = "<%= portal_to_show %>";
  var env_sources_local  = JSON.parse('<%- md_env_package %>');
</script>

<link rel="stylesheet" href="/stylesheets/dhtmlx/dhtmlxtree.css" >

<div class='title ' id='' >OTU Selection</div>


<% if (messages.success) { %>
  	<div class="alert_flash_message_success"><%= messages.success %></div>
<% }else if(messages.fail){ %>
    <div class="alert_flash_message_fail"><%= messages.fail %></div>
<% } %>

      <table class='table' border='1'>
      <tr>

        <td>
           <a  id='clear_filters_btn_id' type='button' value='' class='btn btn-xs btn-primary' >Clear Filters/Reset Projects</a>
        </td>

        <td>
          
          <strong>Select OTU Set Below:</strong></div>
        </td>

        <td>

            <div class='clean_list' >
            
            <li><button type="button" class="btn btn-sm btn-warning" name='Next' id="selection_btn_visuals">
                &nbsp;Visualizations&nbsp;&nbsp;&nbsp;&nbsp;
                <span class="glyphicon glyphicon-circle-arrow-right" aria-hidden="true"></span>
                </button>
                Select how you want to view data.
            &nbsp;&nbsp;&nbsp;
                <% if(user.username == 'guest'){ %>
                    <button type="button" disabled class="btn btn-sm btn-primary" name='Next' id="selection_btn_exports">
                    &nbsp;Download Data&nbsp;
                    <span class="glyphicon glyphicon-circle-arrow-right" aria-hidden="true"></span>
                    </button>
                <% }else{ %>
                    <button type="button" class="btn btn-sm btn-primary" name='Next' id="selection_btn_exports">
                    &nbsp;Download Data&nbsp;
                    <span class="glyphicon glyphicon-circle-arrow-right" aria-hidden="true"></span>
                    </button>
                <% } %>
                Download OTU data file.
            </li>
            </div>
        </td>

      </tr>
      <tr>

        <td style='width:100px;'>
          <% include ../add_ins/otu_filters.html %>
        </td>

        <td>
            <div id='otus_select_div' style='height:500px;overflow:auto;'>
              <span id='nodata_span' style='visiblity:hidden'>Loading Project List</span>
            </div>
        </td>

		<td class='td_top' width="50%">
          <div class='clean_list' style='height:500px;overflow:auto;'>
            
           
           <li>
             The OTUs created for this section have been pre-calculated from VAMPS data as of November, 2010 which includes all of the public 
             <a href='http://icomm.mbl.edu/' target='_blank' rel='noopener'>ICoMM and KCK</a> projects.
             The clusters were generated using the single-linkage preclustering method, SLP, to smooth sequencing errors and reduce noise and an average linkage clustering,
              using clustering thresholds of 3%, 6% and 10%. 
             See <a href='/help/faq#clust' target='_blank'>"How are SLP clusters created"</a> in the FAQ for more information on the clustering method and the SLP citation.
           </li>
            <br>           
           <b>Taxonomy Definition Files:</b>
            <br>The following files define the taxonomy of the new clustering method. 
            Each file contains the representative sequences for that domain, clustered at 3%, 6% or 10%. 
            Here is a full description of the fields in these files. 
            Right-click on the following links to download these files.
            <br>Archaea:	<a href='' onclick="download_data('slp_tax_def','Archaea.slp.pw.an.03.tax')">3%</a> (1.2MB)	<a href='' onclick="download_data('slp_tax_def','Archaea.slp.pw.an.06.tax')">6%</a> (0.7MB)	<a href='' onclick="download_data('slp_tax_def','Archaea.slp.pw.an.10.tax')">10%</a> (0.4MB)
            <br>Bacteria:	<a href='' onclick="download_data('slp_tax_def','Bacteria.slp.pw.an.03.tax')">3%</a> (69.1MB)	<a href='' onclick="download_data('slp_tax_def','Bacteria.slp.pw.an.06.tax')">6%</a> (46.7MB)	<a href='' onclick="download_data('slp_tax_def','Bacteria.slp.pw.an.10.tax')">10%</a> (29.1MB)
            <br>Eukarya:	<a href='' onclick="download_data('slp_tax_def','Eukarya.slp.pw.an.03.tax')">3%</a> (2.3MB)	<a href='' onclick="download_data('slp_tax_def','Eukarya.slp.pw.an.06.tax')">6%</a> (1.0MB)	<a href='' onclick="download_data('slp_tax_def','Eukarya.slp.pw.an.10.tax')">10%</a> (0.4MB)

            <br><br><b>Representative Sequence Files (fasta format): </b>
            <br>The following fasta files (compressed into tar/gzip archives) provide the most abundant sequence in each cluster, which can be used as a representative for that cluster. 

            <br>For these files the fasta definition line includes: 
            <br>1) The cluster ID, 2) an internal ID for a read containing that sequence, 3) the number of reads in that cluster having an exact match to the representative sequence (sequence abundance), and 4) the percent of all reads in that cluster which have an exact match to the representative sequence. These values date from when the clusters were initially created in March 2010.

            <br><a href='' onclick="download_data('slp_repseqs','Archaea.repseqs.tar.gz')">Archaea_repseq</a> (0.3MB)	<a href='' onclick="download_data('slp_repseqs','Bacteria.repseqs.tar.gz')">Bacteria_repseq</a> (19.5MB)	<a href='' onclick="download_data('slp_repseqs','Eukarya.repseqs.tar.gz')">Eukarya_repseq</a> (1.0MB)
            <br><br><b>OTU Membership Files:</b> 
            <br>These files provide the OTU membership in the 3, 6, and 10% OTU clusters for all reads that were clustered. The format includes two columns per line separated by a tab. The first column is the OTU number and the second column is the list of comma-separated reads that comprise that OTU. Each line represents a different OTU.

            <br><a href='' onclick="download_data('slp_otumember','Archaea.otureads.tgz')">Archaea_otu</a> (30.40MB)	<a href='' onclick="download_data('slp_otumember','Bacteria.otureads.tgz')">Bacteria_otu</a> (450.3MB)	<a href='' onclick="download_data('slp_otumember','Eukarya.otureads.tgz')">Eukarya_otu</a> (30.2MB)

          </div>
		</td>

      </tr>
      </table>



<% include ../add_ins/end.ejs %>

<script type="text/javascript" src="/javascripts/global.js" defer></script>
<!-- <script type="text/javascript" src="/javascripts/dhtmlx/dhtmlxtree.js"  defer></script> -->
<!-- <script type="text/javascript" src="/javascripts/load_otu_project_tree.js" defer></script> -->
<script type="text/javascript" src="/javascripts/otus.js" defer></script>
<!-- <script type="text/javascript" src="/javascripts/check_form_pg1.js" defer></script> -->
<!-- <script type="text/javascript" src="/javascripts/project_dataset_tree.js" defer></script> -->

<script defer>

    // code for tooltips
    var $liveTip_pjds = $('<div id="livetip_pjds"></div>').hide().appendTo('body'),
        $win = $(window),
        showTip;

    var tip = {
      title: '',
      offset: 12,
      //delay: 300,
      position: function(event) {
        var positions = {x: event.pageX, y: event.pageY};
        var dimensions = {
          x: [
            $win.width(),
            $liveTip_pjds.outerWidth()
          ],
          y: [
            $win.scrollTop() + $win.height(),
            $liveTip_pjds.outerHeight()
          ]
        };

        for ( var axis in dimensions ) {
          if (dimensions[axis][0] <dimensions[axis][1] + positions[axis] + this.offset) {
            positions[axis] -= dimensions[axis][1] + this.offset;
          } else {
            positions[axis] += this.offset;
          }
        }
        $liveTip_pjds.css({
          top: positions.y,
          left: positions.x
        });

      }
    };


   
</script>
<script defer>
var $, clear_filters, initialize_dhtmlx_otu_tree
$(document).ready(function () {
  // show_dataset_tree = true;
  load_otu_list()
  //clear_filters()
  
})
</script>


<!-- <script type="text/javascript" src="/javascripts/visuals_index.js" defer></script> -->
 </body>
 </html>
