<% include ../add_ins/begin.ejs %>

<%  var items  = JSON.parse(chosen_id_name_hash); %>

<script>
  var ds_local  = JSON.parse('<%- chosen_id_name_hash %>');
  var user_local  = '<%= user.username %>';
</script>
  
  <div class='title' >OTU Method Selection</div>

<a href='' onclick='javascript:window.history.back();return false;'>&lt;&lt;--back</a>

<div class="row">
    <div  id='show_selected_datasets_div'  class="col-md-12">
    	<div class='well'>
                  <% include ../visuals/partials/show_selected_datasets.html %>
      </div>  <!-- /well -->
    </div> <!-- End  show_selected_datasets_div -->

</div>






<form id='' class='' method='POST' action='create_otus_fasta'>
<div class='well'>

	<div class="row">

				<div class="col-md-12">


								<table class='table'>
								<tr><th colspan='2'>OTU Creation Methods</th><th colspan='3'>OTU Size</th><th>
                  <span class='pull-right'>Step-1: <input class='' type='submit' value='Create Fasta File'></span>
                </th></tr>
								<tr>
									<td><input type='radio' name='otu_method' value='closed_ref' onclick="adjust_radios('closed_ref')">
									</td><td><div class='' style='width:300px'>
										Closed Reference OTUs using the Greengenes database	and USEARCH
                    <br><small>External Links: <ul><li><a href='http://www.drive5.com/usearch/' target='_blank'>USEARCH website</a></li>
                    <li><a href='http://greengenes.lbl.gov/cgi-bin/nph-index.cgi' target='_blank'>Greengenes website</a></li></ul></small>
									</div>
									</td>
									<td>3%<br><input type='radio' id='ref_otu_size3' disabled name='ref_otu_size' value='3'></td>
									<td>6%<br><input type='radio' id='ref_otu_size6' disabled name='ref_otu_size' value='6'></td>
									<td>9%<br><input type='radio' id='ref_otu_size9' disabled name='ref_otu_size' value='9'></td>
									<td><div class='' style='height:100px;overflow:auto;'>
										In this reference-based method, each sequence is compared to a reference database of OTU centroids based on the Greengenes sequence database. If a read is within the cluster threshold of a Greengenes centroid, the read is assigned to the OTU represented by that centroid. If the read does not match any reference centroids, it is removed from the analysis.
									</div>
									</td>
								</tr>
								<tr>
									<td>
										<input type='radio' checked name='otu_method' value='uclust' onclick="adjust_radios('uclust')">
										</td><td>
										UCLUST
                    <br><small>External Link: <ul><li><a href='http://drive5.com/usearch/manual/uclust_algo.html' target='_blank'>UCLUST website</a></li>
                    </ul></small>
                  </td>
									<td>3%<br><input type='radio' id='ucl_otu_size3' checked name='ucl_otu_size' value='3'></td>
									<td>6%<br><input type='radio' id='ucl_otu_size6' name='ucl_otu_size' value='6'></td>
									<td>9%<br><input type='radio' id='ucl_otu_size9' name='ucl_otu_size' value='9'></td>
									<td><div class='' style='height:100px;overflow:auto;'>
										A greedy, de novo algorithm, steps through the set of sequences in order of decreasing abundance. The most abundant sequences is the first OTU centroid. Each subsequent sequence is tested against the existing centroids: if it is within the clustering threshold, the sequence is assigned to the OTU represented by that centroid, otherwise the sequence is compared to each of the remaining OTU centroids. If sequence has no matching centroid, it becomes the centroid for a new OTU.
									</div>
									</td>
								</tr>
								<tr>
									<td><input type='radio' name='otu_method' value='slp' onclick="adjust_radios('slp')">
										</td><td>
										SLP - Single-linkage preclustering and Average Linkage clustering using mothur.
									</td>
									<td>3%<br><input type='radio' id='slp_otu_size3' disabled name='slp_otu_size' value='3'></td>
									<td>6%<br><input type='radio' id='slp_otu_size6' disabled name='slp_otu_size' value='6'></td>
									<td>9%<br><input type='radio' id='slp_otu_size9' disabled name='slp_otu_size' value='9'></td>
									<td><div class='' style='height:100px;overflow:auto;'>
										This method preclusters the sequences using a 2% modified-single linkage. The sequences are sorted in order of decreasing abundance. The most abundant sequence becomes the seed for the first OTU. Each subsequent sequence is compared to the sequences in the existing OTUs, the largest first. If the sequence is within 2% of any sequence in an OTU, the sequence is assigned to that OTU, otherwise it is compared to the remaining OTUs. If the sequence fails to fall within 2% of any of the preceding sequences, it becomes the seed for a new OTU. Once preclustering OTUs are created, they are not merged. Once the reads have been preclustered, the most abundant sequence in each precluster is chosen to represent all of the reads assigned to the precluster. Each precluster representative and the size of the precluster are then used as input to a standard average linkage hierarchical clustering using pairwise alignments and distance calculations.
									</div>
									</td>
								</tr>
								<tr>
									<td><input type='radio' name='otu_method' value='crop' onclick="adjust_radios('crop')">
										</td><td>
										CROP - Bayesian Clustering
                    <br><small>External Link: <ul><li><a href='https://github.com/tingchenlab/CROP' target='_blank'>CROP website</a></li>
                    </ul></small>
									</td>
									<td>3%<br><input type='radio' id='crp_otu_size3' disabled name='crp_otu_size' value='3'></td>
									<td>5%<br><input type='radio' id='crp_otu_size6' disabled name='crp_otu_size' value='6'></td>
									<td></td>
									<td><div class='' style='height:100px;overflow:auto;'>
										This is an unsupervised Bayesian method which creates clusters based on the internal organization of the data, rather than relying on inflexible clustering thresholds (e.g., 3%). The clustering is done iteratively to optimize the final cluster assignments.
									</div>
									</td>
								</tr>
							</table>
				</div>

	</div>

</div>

</form>
 <% include ../add_ins/end.ejs %>
  <script type="text/javascript" src="/javascripts/global.js"></script>
  <script type="text/javascript" src="/javascripts/common_selection.js"></script>
  <script type="text/javascript" src="/javascripts/bootstrap-select.js"></script>
 	<script type="text/javascript" src="/javascripts/otus.js"></script>
 </body>
 </html>
