<% menu_section='OVERVIEW' %>
<% include ../add_ins/begin.ejs %>
		<div id='' class='pull-right' style=''>
    		<a href='/help/faq' class='btn btn-primary' target='_blank'>FAQ Page</a>
    </div>
    <p class='title'>Reference Database Files</p>

    

<h2>Reference Database Files</h2>
<br />


<p class="standard">
<b>RefSSU</b>, the primary reference database of near full-length reference sequences, is derived from the 
<a href='http://www.arb-silva.de/' target='_blank' rel='noopener'>SILVA</a>
rRNA database project (version 119). Low quality sequences are flagged as deleted (pintail score <40, sequence quality <50 or alignment quality <50). 
Taxonomic sources include Entrez genomes, the RDP, SILVA, EMBL, and hand-curation by collaborators. A <b>RefSSU_ID</b> is assigned to each sequence 
and is formed from the accession number and the start and stop locations of the sequence on the rRNA gene.
</p>



<p class="standard">
Individual reference files for specific hypervariable regions (<b>RefHVR_v3, RefHVR _v6, RefHVR _v9</b>) are created by excising in-silico the appropriate 
section of the full-length sequences.  Only sequences that cover the entire hypervariable region are included.  A <b>RefHVR_ID</b> is assigned to each unique 
hypervariable region sequence.  RefHVR_IDs are prefixed with the hypervariable region it includes (e.g., v6_AF153).  
The suffix of the ID is simply a unique alphanumeric that contains no additional information. If multiple RefSSU entries have the same hypervariable 
region they will have the same RefHVR_ID for that region.  The database files contain the necessary information to determine the RefSSU source(s) for 
each RefHVR_ID.
</p>


<p class='standard'>
<b>Fasta files</b> include the reference ID, the taxonomy assigned, and the source of the taxonomy.  The RefHVR fastas include both the RefHVR_ID and the source RefSSU_ID.  Only high-quality sequences are included in the fasta files.
</p>
<hr>
<ul>
<p class="standard">RefSSU file:</p>

<li class="standard">RefSSU: full length, unaligned fasta, based on SILVA119: <a href='/static_dnld/refssu.fa.gz'>refssu.fa.gz</a> (125M). 
<br />Aligned sequences are available directly from <a href='http://www.arb-silva.de/' target='_blank' rel='noopener'>SILVA</a>.</li>
<hr>
</ul>
<p class="standard">RefHVR:  SILVA sequences cut between the specified primers:
</p>
<div>
<table id='refhvr_table' border='1' style='border-collapse:collapse;' cellpadding="3">
<tr><th nowrap='nowrap'>SSU rRNA Region</th><th>Primers</th><th>Unaligned Fasta</th><th>GAST Format</th><th>Note</th></tr>
<tr>
  <td style='text-align:center;' center'>v3</td><td>338F - 533R</td>
  <td><a href='/static_dnld/refhvr_v3.fa.gz'>refhvr_v3.fa.gz</a></td>
  <td><a href="/static_dnld/refv3.tgz">refv3.tgz</a></td>
  <td></td>
</tr>
<tr>
  <td style='text-align:center;'>v3v5</td><td>341F - 785Fa**</td>
  <td><a href='/static_dnld/refhvr_v3v5.fa.gz'>refhvr_v3v5.fa.gz</a></td>
  <td><a href="/static_dnld/refv3v5.tgz">refv3v5.tgz</a></td>
  <td></td>
</tr>
<tr>
  <td style='text-align:center;'>v4v6</td><td nowrap='nowrap'>565Fa** - 1064R</td>
  <td><a href='/static_dnld/refhvr_v4v6.fa.gz'>refhvr_v4v6.fa.gz</a></td>
  <td><a href="/static_dnld/refv4v6.tgz">refv4v6.tgz</a></td>
  <td>Assumes 3' - 5' sequencing.</td>
</tr>
<tr>
  <td style='text-align:center;'>v4v6a</td><td nowrap='nowrap'>685Fa** - 1048R</td>
  <td><a href='/static_dnld/refhvr_v4v6a.fa.gz'>refhvr_v4v6a.fa.gz</a></td>
  <td></td>
  <td>Assumes 3' - 5' sequencing.<br>Primer locations optimized for Archaea.</td>
</tr>
<tr>
  <td style='text-align:center;'>v6</td><td>967F - 1064R</td>
  <td><a href='/static_dnld/refhvr_v6.fa.gz'>refhvr_v6.fa.gz</a></td>
  <td><a href="/static_dnld/refv6.tgz">refv6.tgz</a></td>
  <td></td>
</tr>
<tr>
  <td style='text-align:center;'>v6a</td><td>958F 1048R</td>
  <td><a href='/static_dnld/refhvr_v6a.fa.gz'>refhvr_v6a.fa.gz</a></td>
  <td><a href="/static_dnld/refv6a.tgz">refv6a.tgz</a></td>
  <td>Primer locations optimized for Archaea.</td>
</tr>
<tr>
  <td style='text-align:center;'>v9</td><td>1380F - 1510R</td>
  <td><a href='/static_dnld/refhvr_v9.fa.gz'>refhvr_v9.fa.gz</a></td>
  <td><a href="/static_dnld/refv9.tgz">refv9.tgz</a></td>
  <td>Primer locations optimized for Eukarya.</td>
</tr>
<tr>
  <td style='text-align:center;'>refssu</td><td></td>
  <td></td>
  <td><a href='/static_dnld/refssu.tgz'>refssu.tgz</a></td>
  <td>Full length database.</td>
</tr>
</table>
</div>
** the full v3v5 and v4v6 regions are longer than current 454 technology can read.  We trim sequences to an "anchor" location within the read (~400-480nt).  Since these anchor sequences are not primers, their bases are included in the reference files.



<hr>
<ul>
<p class="standard">Template V6 Sequences: Clone43, E. coli, and S. epidermidis.</p>
<li class="standard">Template amplicons sequenced on a 454 Titanium platform: <a href='/static_dnld/IroningWrinkles_fasta.tar.gz'>IroningWrinkles_fasta.tar.gz</a></li>
<li class="standard">Reference:<br> Huse, S. M., D. Mark Welch, H. G. Morrison, and M. L. Sogin. (2010). 
    <a href='http://onlinelibrary.wiley.com/doi/10.1111/j.1462-2920.2010.02193.x/full'>Ironing out the wrinkles in the rare biosphere through improved OTU clustering</a>. Environmental Microbiology 12:1889-1898.
</li>
</ul>

<hr>
<h2>Public Data Files</h2>
<ul>
<p class="standard"></p>
<li class="standard">Public Project Metadata: <a href='/static_dnld/vamps_public_metadata.tsv.gz'>vamps_public_metadata.tsv.gz</a></li>
<li class="standard">Public Project Raw Sequences: <a href='/static_dnld/vamps_public_rawseqs.tgz'>vamps_public_rawseqs.tgz</a></li>
</li>
</ul>


<% include ../add_ins/end.ejs %>
</body>
</html>
