<% include ../add_ins/begin.ejs %>

<%  var items  = JSON.parse(chosen_id_name_hash); %>
<%  var referer = 'export_data'; %>
<script defer>
  var ds_local  = JSON.parse('<%- chosen_id_name_hash %>');
  var user_local  = '<%= user.username %>';
</script>
  <div class='title title_color_exp' >Download Data</div>
  <div class='subtitle'>File Format Selection</div>

<a href='' onclick='javascript:window.history.back();return false;'>&lt;&lt;--back</a>


<form method='POST' name='export_data_form' action='/user_data/export_confirm'>
<div class='well'>

  <div class="row">
    <div  id='show_selected_datasets_div'  class="col-md-6">
                  <% include ../partials/show_selected_datasets.html %>
    </div> <!-- End  show_selected_datasets_div -->
    <div  id=''  class="col-md-6">
                  <div class='' style=''><% include partials/tax_simple.html %></div>
    </div> <!-- End  show_selected_datasets_div -->
  </div>

<% if(messages.success){ %>
  <div class="alert_flash_message_success" style='font-size:large;margin:-20px'><%- messages.success %> </div>
<% }else if(messages.fail){ %>
    <div class="alert_flash_message_fail" style='font-size:large;margin:-20px'><%- messages.fail %> </div>
<% } %>



</div>  <!-- /well -->

<button type='submit' class='btn pull-right btn-sm btn-warning' id="" >
    Get Files <span class="glyphicon glyphicon-circle-arrow-right" ></span>
</button>

**Normalization:
                <input type="radio" checked name='normalization' value="not_normalized"> none&nbsp;&nbsp;
                <input type="radio" name='normalization' value="normalized_to_maximum"> to the Maximum Sample Count&nbsp;&nbsp;
                <input type="radio" name='normalization' value="normalized_by_percent"> by Percent (frequency)
<table class='table'>
  <tr>
    <td>
          <table class='table table-striped table-condensed'>
            <tbody>
              <tr>
                <td>
                 <input type="checkbox" name='fasta' value="1"> Sequences File (Standard Format FASTA)
                </td>
                <td></td>
                <td>Uniqued sequences with raw frequency count in defline: ">readID|dataset|frequency:234"</td>
              </tr>
              <tr>
                <td>
                 <input type="checkbox" name='fastaTAX' value="1"> Sequences File (Standard Format FASTA with Silva/GAST taxonomy and gast_distance) 
                </td>
                <td></td>
                <td>Uniqued sequences: ">readID|dataset|distance|taxonomy|frequency:234"</td>
              </tr>
              <tr>
                <td>
                 <input type="checkbox" name='fastaMED' value="1"> Sequences File (MED or Oligotyping Format FASTA) 
                </td>
                <td></td>
                <td>Non-Uniqued sequences: ">dataset readID-(index)";  Read about: <a href='http://merenlab.org/software/med/'>Minumum Entropy Decomposition (MED)</a> or <a href='http://merenlab.org/software/oligotyping/'>Oligotyping</a></td>
              </tr>
              <tr>
                <td>
                 <input type="checkbox" name='fastaVAMPS' value="1"> Sequences File (VAMPS Format FASTA) 
                </td>
                <td></td>
                <td>Non-Uniqued sequences: ">dataset_(index) readID" (Useful for re-upload to VAMPS)</td>
              </tr>
              
              
              
              <tr>
                <td>
                 <input type="checkbox" name='biom' value="1"> Biom-Format File (JSON)<sup>**</sup>
                </td>
                <td></td>
                <td>See <a href='http://biom-format.org/documentation/format_versions/biom-1.0.html' target='_blank' rel='noopener'>http://biom-format.org/documentation/format_versions/biom-1.0.html</a></td>
              </tr>
              <tr>
                <td>
                <input type="checkbox" name='matrix' value="1"> Taxonomy Matrix File (TaxByTax -- counts matrix)<sup>**</sup>
                </td>
                <td>
                </td>
                <td>Tab-separated. Includes taxonomy (Reads having the same taxonomic assignment are grouped --coarse grouping).</td>
              </tr>
<!--
              <tr>
                <td>
                    <input type="checkbox" disabled name='taxbyref' value="1"> TaxByRef File (<span style='color:red'>not available</span>)
                </td>
                <td></td>
                <td>Reads with the best match to the same reference are grouped (medium grouping).</td>
              </tr>
-->
              <tr>
                <td>
                <input type="checkbox" name='taxbyseq' value="1"> Sequence Matrix File (TaxBySeq -- counts matrix)<sup>**</sup></td>
                <td></td>

                <td>Tab-separated. Includes taxonomy and sequences (Reads having the exact same sequence are grouped --fine grouping).
                </td>
              </tr>
              <!-- 
<tr>
                <td>
                <input type="checkbox" name='taxbytax' value="1"> TaxByTax File (counts matrix)<sup>**</sup></td>
                <td></td>

                <td>Tab-separated. Includes taxonomy ( Reads having the same taxonomic assignment are grouped ~~coarsest grouping).
                </td>
              </tr>
 -->
              <tr>
                <td>
                <input type="checkbox" name='metadata1' value="1"> Metadata File (Datasets/samples as rows)</td>
                <td></td>
                <td>Tab-separated metadata (mapping) file with metadata names as columns</td>
              </tr>

              <tr>
                <td>
                <input type="checkbox" name='metadata2' value="1"> Metadata File (Metadata terms as rows)</td>
                <td></td>
                <td>Tab-separated metadata file with sample names as columns</td>
              </tr>

              
            </tbody>
          </table>
    </td>
  </tr>
</table>


</form>


 <% include ../add_ins/end.ejs %>
  <script type="text/javascript" src="/javascripts/global.js" defer></script>
  <script type="text/javascript" src="/javascripts/common_selection.js" defer></script>
  <script type="text/javascript" src="/javascripts/bootstrap-select.js" defer></script>

 </body>
 </html>
