<% menu_section='MORE' %>
<% include ../add_ins/begin_nomenu.ejs %>


<% if (messages.success) { %>
  	<div class="alert_flash_message_success"><%= messages.success %></div>
<% }else if(messages.fail){ %>
    <div class="alert_flash_message_fail"><%= messages.fail %></div>
<% } %>

<div class="hero-unit">
    <p class='title'>VAMPS FAQ</p>
		<p class='subtitle'>Frequently Asked Questions</p>
</div>

<div  class="row row-offcanvas row-offcanvas-right" >
		 <div  class="col-xs-6 col-sm-3 sidebar-offcanvas pull-right" id="sidebar" style='position:fixed;right: 10px;top:80px'>
		  <div class="list-group">
		  	
		  	<a href="#tag_generation" 	class="list-group-item list-group-item-warning">Tag Generation / Sequence Processing</a>
		  	<a href="#pipeline" 	    class="list-group-item list-group-item-warning">Sequencing Pipeline</a>
		  	<a href="#quality_control" 	class="list-group-item list-group-item-warning">Quality Control</a>
		  	<a href="#gast" 	        class="list-group-item list-group-item-warning">What is GAST?</a>
		  	<a href="#export_data" 		class="list-group-item list-group-item-warning">Exporting (Downloading) Data</a>		  	
		    <a href="#import_data" 		class="list-group-item list-group-item-warning">Importing Data</a>
		    <a href="#clustering" 		class="list-group-item list-group-item-warning">Clustering and Diversity</a>
		    <a href="#phyloseq" 		class="list-group-item list-group-item-warning">What are the Phyloseq Files</a>
		    <a href="#databases" 		class="list-group-item list-group-item-warning">Reference Databases</a>
		    <a href="#submissions" 		class="list-group-item list-group-item-warning">Sample Submission Process</a>
		    
		    <a href="#metadata" 		class="list-group-item list-group-item-warning">Metadata</a>
		    <a href="#formats" 			class="list-group-item list-group-item-warning">Format Definitions</a>
		    <a href="#definitions" 		class="list-group-item list-group-item-warning">Definitions</a>
		    
		  </div>
		</div><!--/.sidebar-offcanvas-->
</div>

<div class="col-lg-9">


      
<!-- TAG GENERATION -->        <a name='tag_generation'></a>
      <hr>
      <div class="row">
        <div class="col-md-12">

          <h4><center>Tag Generation/Sequence Processing</center></h4><a class='btn btn-sm pull-right' href='faq'><span class="glyphicon glyphicon-menu-up" aria-hidden="true">top</span></a>
          <p class='standard'>In October of 2018, we switched from using the Invitrogen Platinum Taq DNA Polymerase High Fidelity (Cat. No. 11304102) 
          to the Invitrogen Platinum SuperFi DNA Polymerase (Cat. #12351-050). High Fidelity polymerase offers 6X fidelity vs Taq, whereas SuperFi 
          polymerase offers 100X fidelity vs Taq. Modified PCR master mix recipe is as follows:
          </p> 
            <a name='illumina'></a>
            <strong>Illumina Amplicon Generation</strong>

            <br><br><u>Fusion PCR recipe (125 uL):</u>
            <ul>
            <li class="standard">91.25 uL of water</li>
            <li class="standard">25 uL of 5X SuperFi Buffer</li>
            <li class="standard">2.5 uL of 10 mM dNTP Mix</li>
            <li class="standard">1.25 uL of Platinum SuperFi Polymerase</li>
            <li class="standard">4 uL of 10 uM Fusion Primer Mix</li>
            <li class="standard">1 uL of template*</li>
            </ul>
            *More template can be used in place of water if sample concentration is low (< 1 ng).
            <br>
            Prior to addition of template, 25 uL of master mix is removed to serve as a negative control. After addition of template, master mix is divided into 3 reactions of 33 uL each.
            <br><br>
            <u>v4v5 Program:</u>
            <ul>
            <li class="standard">94&deg;C for 3 min</li>
            <li class="standard">30 cycles of:</li>
                <ul>
                <li class="standard">94&deg;C for 30 sec</li>
                <li class="standard">57&deg;C for 45 sec</li>
                <li class="standard">72&deg;C for 1 min</li>
                </ul>
            <li class="standard">72&deg;C for 2 min</li>
            <li class="standard">Hold at 4&deg;C</li>
            </ul>
            <u>Supplies:</u>
            <br>Invitrogen Platinum SuperFi DNA Polymerase (Cat. #12351-050)
            <br>Thermo Scientific 10 mM dNTP Mixes (Cat. # FERR0192)
            
<hr> 
             <p class='standard'>
				     In 2013 we switched to using <strong>Illumina</strong> platforms for 16S sequencing. Similar to the strategy for 454, we use fusion primers composed
				     of the Illumina adaptors, multiplexing identifiers, and domain-specific primers.
				     Thermocycling and reaction mixtures are different from 454 sequencing for Illumina amplicon PCR.
				    </p>
				    <p class='standard'>
				       <strong>For Archaeal V6:</strong></p>

				    <ul>
				        <li class="standard">94C for 2 min</li>
				        <li class="standard">30 cycles of (94C for 30sec, 60C for 45sec, 72C for 1min)</li>
				        <li class="standard">72C for 2 min</li>
				        <li class="standard">Hold at 4C</li>
				    </ul>

				    <p class='standard'>
				       <strong>For Bacterial and Archaeal V4V5:</strong></p>

				    <ul>
				        <li class="standard">94C for 2 min</li>
				        <li class="standard">30 cycles of (94C for 30sec, 57C for 45sec, 72C for 1min)</li>
				        <li class="standard">72C for 2 min</li>
				        <li class="standard">Hold at 4C</li>
				    </ul>

				    <p class='standard'>
				    Bacterial V6 uses first the domain-specific primers for 25 cycles, then the products are cleaned and used in a second 5-cycle PCR with fusion primers.
				    </p>
				    <p class='standard'>
				       <strong>For Bacterial V6:</strong></p>

				    <ul>
				        <li class="standard">94C for 2 min</li>
				        <li class="standard">25 cycles of (94C for 30sec, 60C for 45sec, 72C for 1min)</li>
				        <li class="standard">72C for 2 min</li>
				        <li class="standard">Hold at 4C</li>
				        <li class="standard">Visualize and quantitate the amplicons in the BioAnalyzer or Caliper.</li>
				        <li class="standard">Purify the amplicons using Qiagen MinElute columns.</li>
				        <li class="standard">Use product in second PCR:</li>
				        <ul>
				          <li class="standard">94C for 2 min</li>
					      	<li class="standard">5 cycles of (94C for 30sec, 60C for 45sec, 72C for 1min)</li>
					      	<li class="standard">72C for 2 min</li>
					      	<li class="standard">Hold at 4C</li>
					    	</ul>
						</ul>
				    <p class='standard'>
				    After we produce amplicons, we can clean and/or size-select for the target products using Agencourt AMPure XP beads.
				    Then we quantitate products with an Invitrogen Picogreen assay, pool at desired concentrations (e.g. equimolar), and quantitate the final pool with qPCR.
				    </p>

<hr>                   
          <a name='454'></a>
          <p class='standard'>Conserved sequences that flank the hypervariable V6-V4 region of rRNAs serve as primer sites to generate PCR
				   amplicons. Each PCR reaction produces products that can be informatically identified using a unique "key" incorporated between
				   the 454 Life Sciences primer A or B and the 5' flanking rRNA primer. The use of a 5-bp key allows for the synthesis of as many as 81
				   oligonucleotides that differ by at least two sites. Our multiplexing strategy allows the concurrent collection of 10,000-50,000
				   tags from each of 8-40 samples in a single nine- hour sequencing run without use of partitioning gaskets that reduce the number
				   of sequencing wells on the PicoTiterPlate<small><sup>TM</sup></small>. Amplicons can be pooled before the emPCR step and each
				   pool is run on a large region of the plate.
				   </p>
				   <p class='standard'>
				       <strong>454 Amplicon PCR</strong> (Christina Holmes/Ekaterina Andreishcheva) for four reactions:</p>
				       <ul>
				         <li class="standard">96 ul water</li>
				        <li class="standard">13.4 ul 10X Platinum buffer</li>
				        <li class="standard">9.4 ul 50 mM MgSO4</li>
				        <li class="standard">2.6 ul 10 mM Pure Peak dNTPs</li>
				        <li class="standard">4 ul 10uM Fusion Primer A </li>
				        <li class="standard">4 ul 10uM Fusion Primer B</li>
				        <li class="standard">2.6 ul 2.5 U/ul Platinum HiFi Pol</li>
				        <li class="standard">[2 ul template (~5-25 ng)*]</li>
				        <li class="standard">33 ul total volume/reaction</li>
				       </ul>

				     <p class='standard'>*If template stock is dilute or otherwise resistant to amplification, more template can be added in place of water.
				     </p>
				     <p class='standard'>
				     The 5 reactions are the three replicates of the environmental template, positive control, and negative control.
				     Template (plasmid pool for positive control; water for negative control) is added as final step.
				     </p>
				     <p class='standard'>
				       <strong>Program:</strong></p>
				       <ul>
				        <li class="standard">94C for 2 min</li>
				        <li class="standard">30 cycles of (94C for 30sec, 60C for 45sec, 72C for 1min)</li>
				        <li class="standard">72C for 2 min</li>
				        <li class="standard">Hold at 4C</li>
				        <li class="standard">Visualize and quantitate the amplicons in the BioAnalyzer or Caliper.
				        <li class="standard">Purify the amplicons using Agencourt AMPure XP beads as described in 454 Sequencing Technical Bulletin 2011-007, and resuspend in 100 uL of Buffer EB.</li>
				        <li class="standard">Store purified amplicons at -20C.</li>
				       </ul>
				     <br />

				      <p class='standard'><strong>Supplies:</strong></p>
				     <ul>
				        <li class="standard">DNA 1000 Kit, Agilent, 5067-1504, 25 chips. $372</li>
					    	<li class="standard">Agencourt AMPure XP, 60 mL, A63881. $1050</li>
					    	<li class="standard">PurePeak DNA Polymerization Mix 10 mM 1 ml, Pierce/ThermoFisher, NU606001, $115</li>
					    	<li class="standard">Platinum HiFi Taq polymerase plus buffer and MgSO4, Invitrogen.</li>
				     </ul>

				   
        </div>
      </div>
<!-- PIPELINE -->        <a name='pipeline'></a>
      <hr>
      <div class="row">
        <div class="col-md-12">
            <h4><center>Sequencing Pipeline</center></h4><a class='btn btn-sm pull-right' href='faq'><span class="glyphicon glyphicon-menu-up" aria-hidden="true">top</span></a>
            <p><strong>Our sequencing pipeline is public and the details are stored on GitHub. The links are here:</strong></p>
            <ul>
            <li><a href='https://github.com/annaship/py_mbl_sequencing_pipeline' target='_blank' rel='noopener'>https://github.com/annaship/py_mbl_sequencing_pipeline</a>
            <li><a href='https://github.com/annaship/py_mbl_sequencing_pipeline/blob/master/pipeline/illumina_files.py' target='_blank' rel='noopener'>https://github.com/annaship/py_mbl_sequencing_pipeline/blob/master/pipeline/illumina_files.py</a>
            </ul>
            
            <p>The Steps in order are: Demultiplexing, Merging, Uniqueing, Chimera Checking and Taxonomy Assignment</p>
    
            <p>We use <a href='http://merenlab.org/' target='_blank' rel='noopener'>Meren's</a> scripts to perform some of these steps. You can find them here on his GitHub site: <a href='https://github.com/merenlab/illumina-utils/tree/master/scripts' target='_blank' rel='noopener'>https://github.com/merenlab/illumina-utils/tree/master/scripts</a></p>
            <ol>
                <li><strong>Demultiplexing</strong></li>
                <ul>
                  <li><a href='https://github.com/merenlab/illumina-utils/blob/master/scripts/iu-demultiplex' target='_blank' rel='noopener'>https://github.com/merenlab/illumina-utils/blob/master/scripts/iu-demultiplex</a></li>
                </ul>
                
                
                <br><li><strong>Merging (with <a href='#quality_control'>Quality Control</a>)</strong></li>
                <ul>
                  <li>From <a href='https://github.com/merenlab/illumina-utils/tree/master/scripts' target='_blank' rel='noopener'>https://github.com/merenlab/illumina-utils/tree/master/scripts</a></li>
                  <ul>
                    <li>overlap: iu-merge-pairs</li>
                    <li>filter mismatches: iu-filter-merged-reads</li>
                    <li>for v6 only: iu-trim-V6-primers</li>
                  </ul>
                </ul>
                
                
                <br><li><strong>Uniqueing</strong></li>
                <ul>
                <li><a href='https://gist.github.com/meren/abd5c6f014da9aa554a4' target='_blank' rel='noopener'>https://gist.github.com/meren/abd5c6f014da9aa554a4</a></li>
                </ul>
                
                
                <br><li><strong>Chimera Checking</strong></li>
                <ul>
                  <li>We use <a href='https://github.com/torognes/vsearch' target='_blank' rel='noopener'>vsearch</a> for chimera checking (combining result from both ref and denovo)</li>
                 
                    <li>[<b>sample ref:</b>] vsearch -uchime_ref ACAGTG_ACTGC_1_MERGED-MAX-MISMATCH-3.unique.chg -db rRNA16S.gold.fasta -uchimeout ACAGTG_ACTGC_1_MERGED-MAX-MISMATCH-3.unique.chimeras.db -chimeras ACAGTG_ACTGC_1_MERGED-MAX-MISMATCH-3.unique.chimeras.db.chimeric.fa -strand plus -notrunclabel</li>
                    <li>[<b>sample denovo:</b>] vsearch -uchime_denovo TTAGGC_NNNNCGACG_1_MERGED-MAX-MISMATCH-3.unique.chg -uchimeout TTAGGC_NNNNCGACG_1_MERGED-MAX-MISMATCH-3.unique.chimeras.txt -chimeras TTAGGC_NNNNCGACG_1_MERGED-MAX-MISMATCH-3.unique.chimeras.txt.chimeric.fa -notrunclabels</li>
                    
                </ul>
                
                
                <br><li><strong>Taxonomy Assignment (using GAST)</strong></li>
                <ul>
                <li>gast_ill (<a href='https://github.com/annaship/BPC-scripts/blob/master/gast_ill' target='_blank' rel='noopener'>https://github.com/annaship/BPC-scripts/blob/master/gast_ill</a>)</li>
                <li>You also will need <a href='https://github.com/annaship/BPC-scripts/blob/master/Taxonomy.pm' target='_blank' rel='noopener'>Taxonomy.pm</a> and taxonomy reference database files.</li>
                </ul>
            </ol>
        </div>
      </div>
<!-- QUALITY CONTROL -->        <a name='quality_control'></a>
      <hr>
      <div class="row">
        <div class="col-md-12">
            <h4><center>Quality Control</center></h4><a class='btn btn-sm pull-right' href='faq'><span class="glyphicon glyphicon-menu-up" aria-hidden="true">top</span></a>
            <p><strong></strong></p>
            <ul>
              <li>There are several steps in the quality control process. Some of these steps are tied up in other pipeline steps (See <a href='#pipeline'>Sequencing Pipeline</a> above).</li>
              <li>While demultiplexing we check if barcodes and indexes are the same as in provided metadata, if not reads are discarded.</li>
              <li>Merging read 1 and read 2 into fasta files. At this point we also discard low quality reads. Primers and barcodes are striped out.</li>
              <li>For part of it "iu-merge-pairs" and "iu-filter-merged-reads" <a href='http://merenlab.org/' target='_blank' rel='noopener'>Meren's</a> utilities are used. (See <a href='#pipeline'>Sequencing Pipeline</a> above)</li>
              <ul>
                      
             
             <li>Filters used during merging (sequence will be discarded if it fails any of these tests):</li>
                <ol>
                  <li>Is the prefix correct for read_1 and read_2?</li>
                  <li>P value? (P value is the ratio of the number of mismatches and the length of the overlap. Merged sequences can be discarded based on this ratio)</li>
                  <li>Does the read have less than the maximum number of mismatches?</li>
                  <li>Does the read have any 'N's?</li>
                  <li>Better than Q30? (Phred quality score, if Phred assigns a Q score of 30 (Q30) to a base, this is equivalent to the probability of an incorrect base call 1 in 1000 times)</li>
                  <li>More than minimum expected overlap?</li>

                </ol> 
              <br>    
              <li>Default filter values:</li>
                <pre>
"P" value .................................................   0.300000
Maximum number of mismatches in the overlapped region ......  None
Minimum overlap size .......................................  15
Minimum Q-score for mismatches .............................  15
Q30 enforced? ..............................................  True
                </pre>          
              
              </ul>
            <li>Chimera checking using vsearch, both denovo and reference based.</li>  
            </ul>
        </div>
      </div>
      
<!-- GAST -->       <a name='gast'></a>
      <hr>
      <div class="row">
        <div class="col-md-12">
          <h4><center>What is GAST?</center></h4><a class='btn btn-sm pull-right' href='faq'><span class="glyphicon glyphicon-menu-up" aria-hidden="true">top</span></a>
          <p><strong>GAST stands for <u>G</u>lobal <u>A</u>lignment for <u>S</u>equence <u>T</u>axonomy</strong>.</p>
          <p>It uses a reference database of SSU sequences to determine the taxonomy of hypervariable region tags. The specifics are described in the citation below.</p>
          <ul>
          <li><a href='https://github.com/annaship/BPC-scripts/blob/master/gast_ill' target='_blank' rel='noopener'>GAST Code on GitHub</a></li>
          <li><a href='https://github.com/annaship/BPC-scripts/blob/master/Taxonomy.pm' target='_blank' rel='noopener'>Taxonomy.pm perl module</a></li>
          <li>Available GAST-formatted SSU <a href='https://vamps2.mbl.edu/resources/reference_data' target='_blank' rel='noopener'>Reference Sets</a>.</li>
          </ul>
          
          Citation:
          <a href='https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1000255' target='_blank' rel='noopener'>
          <pre>
Huse SM, Dethlefsen L, Huber JA, Welch DM, Relman DA, Sogin ML (2008) 
Exploring Microbial Diversity and Taxonomy Using SSU rRNA Hypervariable Tag Sequencing. 
PLoS Genet 4(11): e1000255. https://doi.org/10.1371/journal.pgen.1000255        
          </pre>
          </a>
        </div>
      </div>      
      
<!-- EXPORT  DATA -->       <a name='export_data'></a>
      <hr>
      <div class="row">
        <div class="col-md-12">
          <h4><center>Exporting (Downloading) Data</center></h4><a class='btn btn-sm pull-right' href='faq'><span class="glyphicon glyphicon-menu-up" aria-hidden="true">top</span></a>
          <p><strong>There is no raw sequence data available on VAMPS.</strong></p>
          <p>The primary way to export data from VAMPS is to use the 'Download Data' page which
          is accessed from the <a href='/visuals/visuals_index'>main 'Sample Selection' page</a>. After you've selected the datasets you want to include in your download
          select the blue 'Download Data' button.
          <br>You can download data from VAMPS in various formats. Listed on the download page are the formats (with descriptions) available to you.
          <ul>
          <li><strong>Fasta</strong> (3 formats available)
          <li><strong>Counts Matrix</strong> (2 formats available)
          <li><strong>Metadata</strong> (2 formats available)
          <li><strong>Biom</strong> (Modified JSON format)
          
          </ul>
            </p>
            
        
        There are other downloading opportunities available for data and images on the Visualization pages.

        </div>
      </div>

 <!-- IMPORT DATA -->      <a name='import_data'></a>
      <hr>
      <div class="row">
        <div class="col-md-12">
          <h4><center>Importing Data</center></h4><a class='btn btn-sm pull-right' href='faq'><span class="glyphicon glyphicon-menu-up" aria-hidden="true">top</span></a>
          <p class='standard'><strong>Description of Import Options</strong></p>
          <a href='/user_data/import_choices'>To upload data to VAMPS go here.</a>
          <br><br><p>Fasta files and count matrix files can be uploaded to VAMPS as long as they conform to the correct format which is described
          on the upload page for each type.           
          </p>
          
          <p class='standard'><strong>Multiple-Dataset fasta file</strong></p>
          <p>
          </p>
          <p class='standard'><strong>Metadata file (for projects already in VAMPS)</strong></p>
          <p>This is a csv (comma separated values) file but the data are actually separated by &lt;TABS&gt; to make it more human readable.
          	These files must conform to the <a href='http://qiime.org/documentation/file_formats.html' target='_blank' rel='noopener'>qiime mapping file format</a>.
          	Since these are processed (trimmed) data that are being uploaded, the 'BarcodeSequence' and 'LinkerPrimerSequence' fields can be left empty
          	but the header names have to be present. The #SampleID field must be present and there can be no duplicate sample names.
          	The Description field must also be present and must be the last field. There is a handy 'validate_mapping_file.py' script available
          	in qiime to assist you with this file.
          </p>
                  
          
        </div>
      </div>
      
 <!-- CLUSTERING -->      <a name='clustering'></a>
      <hr>
      <div class="row">
        <div class="col-lg-12">
          <h4><center>Clustering and Diversity</center></h4><a class='btn btn-sm pull-right' href='faq'><span class="glyphicon glyphicon-menu-up" aria-hidden="true">top</span></a>

          <strong><a name="clust">How are SLP clusters created?</a></strong> 
        <br />
        <br />
        A combination of ESPRIT, SLP and mothur computes taxonomic independent clusters (Operational Taxonomic Units - OTUs) <br />
        using the total collection of available V6 sequences in VAMPS. The sequences were binned into separate datasets for the <br />
        Archaeal or Eukaryal domains, and into Bacterial phylum- or Proteobacterial class-level datasets.  For each bin, the unique.seqs <br />
        function in mothur, eliminated duplicate sequences but retained information about observed frequencies for each unique read. <br />
        The kmerdist module of ESPRIT (with default values) identified all sequence pairs within each bin that are predicted to be at<br /> 
        least 90% similar.  The needledist module in ESPRIT generated a sparse matrix of pairwise distances by performing a Needleman-Wunsch <br />
        alignment on the sequence pairs and calculating pairwise distances using quickdist. The algorithm SLP uses the pairwise distances<br /> 
        to perform a modified single-linkage preclustering at 2% to reduce noise in the sequence data.  Initially SLP orders sequences <br />
        according to their rank abundance and then steps through the ordered sequences assigning them to clusters.  The most abundant<br /> 
        sequence defines the first cluster.   Each subsequent sequence is tested against the growing list of clusters using the <br />
        single-linkage algorithm.  If the sequence has a pairwise distance less than 0.02 (equivalent to a single difference in the V6 region)<br /> 
        to any of the sequences already in the cluster, the new sequence will be added to the cluster and not tested against subsequent clusters.<br />  
        If the sequence is not within a distance of 0.02 from any read in any of the existing clusters, it will establish a new cluster. <br />
        Once all sequences have been assigned to clusters, sequences in the low abundance clusters (< 10 tags) are tested against <br />
        the larger clusters and added to those clusters if possible.  For each precluster, SLP uses the sequence with the highest <br />
        frequency and the count of all tags in the precluster for average linkage clustering by mothur. Taxonomy for each cluster <br />
        relies upon on a two-thirds majority of the taxonomy for each cluster member; CATCHALL estimates the estimate richness. 
<br><br>Citation:
<pre>
  Huse, S.M., D. Mark Welch, H.G Morrison, and M.L. Sogin. (2010) <a href='https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2909393/' target='_blank' rel='noopener'>
  Ironing out the wrinkles in the rare biosphere</a>. <i>Environmental Microbiology</i> early view.
</pre>
        </div>
      </div>
<!-- PHYLOSEQ -->      <a name='phyloseq'></a>
	<hr>
      <div class="row">
        <div class="col-md-12">
          <h4><center>Phyloseq and ggplot2</center></h4><a class='btn btn-sm pull-right' href='faq'><span class="glyphicon glyphicon-menu-up" aria-hidden="true">top</span></a>

					<p class='standard'><strong>What is Phyloseq?</strong></p>
					<p>Phyloseq is an R library for microbiome data: (<a href='https://joey711.github.io/phyloseq/' target='_blank' rel='noopener'>https://joey711.github.io/phyloseq/</a>).
						<br>VAMPS uses R (<a href='https://www.r-project.org/' target='_blank'>https://www.r-project.org/</a>) and Python (<a href='https://www.python.org/' target='_blank' rel='noopener'>https://www.python.org/</a>) scripts to produce some of the visualizations.
						<br><br>
						To use the Phyloseq library with your VAMPS data download the three Phyloseq files from the 'Display Choices' Page:
						<br>Import them directly into R (or R-script) as shown below to create a Phyloseq Object:
						<ul>
							<li>biom_file <- 'phyloseq.biom'
							<li>tax_file <- 'phyloseq_taxonomy.txt'
							<li>map_file <- 'phyloseq_metadata.txt'

							<li>library(phyloseq)
							<li>library(ggplot2)
							<li>TAX<-as.matrix(read.table(tax_file, header=TRUE, sep = "\t", row.names = 1, as.is=TRUE))
							<li>OTU <- import_biom(biom_file)
							<li>MAP <- import_qiime_sample_data(map_file)
							<li>TAX <- tax_table(TAX)
							<li>OTU <- otu_table(OTU)
							<li>physeq <- phyloseq(OTU,TAX,MAP)
						</ul>
					</p>
					<p>See the <a href='https://joey711.github.io/phyloseq/' target='_blank' rel='noopener'>Phyloseq website</a> for more help and examples.</p>
					
					<br><p class='standard'><strong>What is ggplot2</strong></p>
					ggplot2 is an R library that provides quality graphic displays using various big data formats such as VAMPS downloads.
					<br>See <a href='https://ggplot2.tidyverse.org/' target='_blank' rel='noopener'>https://ggplot2.tidyverse.org/</a> for more information.
					 
					
        </div>
      </div>
      
      
<!-- DATABASES -->      <a name='databases'></a>
      <hr>
      <div class="row">
        <div class="col-md-12">
          <h4><center>Reference Databases</center></h4><a class='btn btn-sm pull-right' href='faq'><span class="glyphicon glyphicon-menu-up" aria-hidden="true">top</span></a>

					<p class='standard'><strong>How are the reference databases created? </strong></p>
					<p>We create Ref16S, a reference database of aligned full-length sequences based on all available sequences
					in SILVA exported using the ARB software. New updates to both SILVA and RDP are incorporated as they become available.
					<ul>
					<li>We flag as low-quality and delete all sequences with a sequence quality score &lt;= 50, an alignment
					score &lt;= 50 or a pintail (chimera) score &lt;= 40.
					<li>We flag as redundant and delete all exact copies of the full-length sequence.
					<li>We classify all bacterial and archaeal sequences directly with the Ribosomal Database Project Classifier
					(RDP). We used only RDP classifications with a bootstrap value of >=80%. If the bootstrap value was &lt;80%,
					the taxonomic assignment was moved to a higher classification level until an 80% or better bootstrap value was
					achieved. For example, if the genus assignment had a bootstrap value of 70%, but the family had a value of 85%,
					that sequence would be assigned only as far as family and not to genus. RDP Classifier does not classify
					sequences below the genus level.

					<li>We incorporate other taxonomy sources, such as Entrez Genome accession numbers or researcher knowledge of
					specific entries, as they become available. These "other sources" are used preferentially over the RDP for
					bacteria and archaea. RDP does not classify eukaryotes. For eukaryota taxonomies, we use the EMBL taxonomy
					from the SILVA database where we do not have other sources.
					<br>
					<li>We create hypervariable region specific databases (RefV6, RefV3, RefV9, etc.).
					<ul>
					<li>For each hypervariable region we calculate the Ref16S alignment coordinates.
					<li>We then excise from the Ref16s aligned sequences the section corresponding to the hypervariable region.
					<li>The gaps are removed from the aligned sequences to create a set of unaligned sequences.
					Any hypervariable sequences that contain an 'N' are deleted.
					Any hypervariable sequences shorter than 50 nt are deleted.
					Any full-length sequences that were not sequenced all the way through the specific hypervariable region are deleted.
					Two reference IDs are assigned to each reference hypervariable region. A ref16s_id (previously alt_local_gi)
					links the hypervariable sequence with its source full-length sequence, and a second, e.g., refv6_id
					(previously known as local_gi) is used to identify all entries having the exact same sequence of the
					hypervariable region. The taxonomy is carried directly from the full-length source.
					<li>Unique reference hypervariable regions sequences are exported to a blastable database for use in the
					assignment of taxonomy to pyrosequencing reads through GAST.
					</ul>
					</ul>
					</p>

        </div>
      </div>
<!-- SUBMISSIONS -->        <a name='submissions'></a>
      <hr>
      <div class="row">
        <div class="col-md-12">
          <h4><center>Sample Submission Process</center></h4><a class='btn btn-sm pull-right' href='faq'><span class="glyphicon glyphicon-menu-up" aria-hidden="true">top</span></a>
          <p>Donec id elit non mi porta gravida at eget metus. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus. Etiam porta sem malesuada magna mollis euismod. Donec sed odio dui.
          </p>
        </div>
      </div>


<!-- METADATA -->        <a name='metadata'></a>
      <hr>
      <div class="row">
        <div class="col-md-12">
          <h4><center>Metadata</center></h4><a class='btn btn-sm pull-right' href='faq'><span class="glyphicon glyphicon-menu-up" aria-hidden="true">top</span></a>
          <p>Required metadata fields:
          	<div>
          	<table class='table'>
          		
          		<tr><td>collection_date</td><td>Example: 2003-03-25<a class='pull-right' href='/user_data/required_metadata_options'>More Information</a></td></tr>
          		<tr><td>geo_loc_name</td><td>name of country or longhurst zone<a class='pull-right' href='/user_data/required_metadata_options'>More Information</a></td></tr>
          		<!-- 
<tr><td>Country</td><td>Choose from GAZ Ontology.<a class='pull-right' href='/user_data/required_metadata_options'>More Information</a></td></tr>
          		
 -->
          		<tr><td>dna_region</td><td><a class='pull-right' href='/user_data/required_metadata_options'>More Information</a></td></tr>
          		<tr><td>domain</td><td><a class='pull-right' href='/user_data/required_metadata_options'>More Information</a></td></tr>
          		
          		<tr><td>env_biome; env_feature; env_material</td><td><a href='http://www.ebi.ac.uk/ols/ontologies/envo' target='_blank' rel='noopener'>Envo Ontology Browser</a> Biome: Description of the site;
          			<br>Feature: Description of feature in the biome where sample was obtained;<a class='pull-right' href='/user_data/required_metadata_options'>More Information</a>
          			<br>Material: Description of material</td></tr>
          		<tr><td>env_package</td><td><a class='pull-right' href='/user_data/required_metadata_options'>More Information</a></td></tr>
          		<tr><td>target_gene</td><td>Enter '16s' or '18s' for the section of the Small Subunit rRNA<a class='pull-right' href='/user_data/required_metadata_options'>More Information</a></td></tr>
          		<tr><td>latitude</td><td>Geographical origin of sample in decimal degrees (WGS84  system), not DMS</td></tr>
          		<tr><td>longitude</td><td>Geographical origin of sample in decimal degrees (WGS84  system), not DMS</td></tr>
          		<tr><td>sequencing_platform</td><td>'454', 'illumina', 'ion-torrent', 'sanger' or 'unknown'<a class='pull-right' href='/user_data/required_metadata_options'>More Information</a></td></tr>
          		<tr><td>adapter_sequence</td><td>illumina specific</td></tr>
          		<tr><td>illumina_index</td><td>illumina specific</td></tr>
          		<tr><td>primer_suite</td><td>MBL Specific</td></tr>
          		<tr><td>run</td><td>MBL Specific</td></tr>
          	</table>
          </p>
        </div>
      </div>
			</div>
<!-- FORMAT DEFINITIONS -->        <a name='formats'></a>
      <hr>
      <div class="row">
        <div class="col-md-12">

          <h4><center>File and Name Formats</center></h4><a class='btn btn-sm pull-right' href='faq'><span class="glyphicon glyphicon-menu-up" aria-hidden="true">top</span></a>
          <p>

						<p class='standard'><strong>Project Name</strong></p>
						No spaces allowed!

						<p class='standard'><strong>Dataset Name</strong></p>
						No spaces allowed!

						<p class='standard'><strong>FASTA File</strong></p>
						<p class='standard'><strong>Metadata File</strong></p>
						There is one metadata file format allowed for import (except taxbyseq --see below):<br>
						<ul>
						<li>QIIME style mapping file: This file MUST have a header named '#SampleID', 'sample_name',  'dataset' or 'dataset_name' as the <strong>first</strong> column name.
							<ul>
								<li>The Header line is required</li>
								<li>The columns must be tab delimited (unless commas are required by the specific page)</li>
								<li>There must be one row for each dataset</li>
							</ul>
						</li>

						</ul>
						<br>
						<p class='standard'><strong>TaxBySeq File and Metadata from Old (legacy) VAMPS</strong></p>
							<ul>
								<li>See <a href='https://vamps.mbl.edu/exporter/export_data.php' target='_blank' rel='noopener'>VAMPS Exports</a></li>
								<li>Download the TaxBySeq and Metadata files to your harddrive.</li>
								<li>Import them as a pair either directly (still compressed) or uncompressed.</li>
							</ul>
                        <br>
						<p class='standard'><strong>JSON Configuration File</strong></p>
							<ul>
								<li>Required to be valid JSON (try <a href='https://jsonlint.com' target='_blank' rel='noopener'>https://jsonlint.com</a>)</li>
								<li>Requires "source":"VAMPS*"</li>								
								<li>Requires id_name_hash.ids: [ ] --A valid list of dataset ids that you have permissions to view.</li>
								<li>Optional:
								    <ol>
								        <li>normalization&nbsp;&nbsp;(Valid values: "none", "maximum", "frequency")
								        <li>selected_distance&nbsp;&nbsp;(Valid values: "morisita-horn", "jaccard", "kulczynski", "canberra" or "bray-curtis")
								        <li>tax_depth&nbsp;&nbsp;(Valid values: one of "domain", "phylum", "klass", "order", "family", "genus", "species", "strain")
								        <li>include_nas&nbsp;&nbsp;(Valid values: "yes" or "no")
								        <li>domains&nbsp;&nbsp;(Valid values: any or all of "Archaea","Bacteria","Eukarya","Organelle","Unknown")
								        <li>min_range&nbsp;&nbsp;(Valid values: integers 0-99)
								        <li>max_range&nbsp;&nbsp;(Valid values: integers 1-100)								        
								    </ol>
								<li>Image -- (optional) To render a table, chart or other visual element automatically on upload try:</li>
								    <ul>
								    "image":"dheatmap"  (added at same level as "source")
								    <br>possible values: "dheatmap", "piecharts", "barcharts", "counts_matrix", "metadata_table", "fheatmap", "dendrogram01", "dendrogram03", "pcoa", "pcoa3d",
								    "geospatial" or "adiversity"
								    </ul>
								</li>
							<pre>
SAMPLE:							
{
  "source":"VAMPS",
  "post_items":
  { 
    "normalization":"maximum",
    "selected_distance":"morisita_horn",
    "tax_depth":"phylum",
    "domains":["Archaea","Bacteria","Eukarya","Organelle","Unknown"],
    "include_nas":"yes",
    "min_range":0,
    "max_range":100
  },
    "id_name_hash":
    {
      "ids":["49","50","51","52"]
    }
}
							</pre>
							
							</ul>
							
          </p>
        </div>
      </div>
<!-- DEFINITIONS -->        <a name='definitions'></a>
      <hr>
      <div class="row">
        <div class="col-md-12">

          <h4><center>Definitions</center></h4><a class='btn btn-sm pull-right' href='faq'><span class="glyphicon glyphicon-menu-up" aria-hidden="true">top</span></a>
          <p>

						<p class='standard'><strong>Project</strong></p>

						The project name refers to the overall study or research project to which the data belong.
						The project ties multiple samples and sequencing runs together.

						<p class='standard'><strong>Dataset</strong></p>

						The dataset name refers to a set of sequences within the project that are from one sampling
						location or individual at a particular date and time. The dataset combines sequences sampled or amplified together.
						Sequence and taxonomic data are uploaded on a dataset by dataset basis. Multiple datasets may be combined together
						or compared separately when using the Community Visualization tools.

						<p class='standard'><strong>FASTA Files</strong></p>
						When you upload a file it will be filtered for valid file format and data.
						If valid, the file will be uploaded into a temporary table of VAMPS data that will be available immediately for viewing.
						<p class='standard'><strong>FASTA definition line (or defline)</strong></p> The FASTA file defline follows NCBI FASTA format. Each read starts with a ‘&gt;’
						and the read ID is between the ‘&gt;’ and the first ‘|’
						(a ‘pipe’ symbol), it cannot contain any special characters other than dash ‘-’ or underscore ‘_’ and must be less than 32 characters.
						If there is any other information on the definition line, it must be after the first ‘|’. The whole definition line is separated from the sequence data by a return or linefeed.


          </p>
        </div>
      </div>



<% include ../add_ins/end.ejs %>

</div><!--/.col-lg-9-->


 </body>
 </html>
